org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/launch/BaseLaunchRequestJob.java

```diff
@@ -137,6 +139,16 @@ private String toNonEmpty(List<String> stringValues, String defaultMessage) {
         return string != null ? string : defaultMessage;
     }
 
+    private BuildConfiguration buildConfigurationFor(RunConfiguration runConfig) {
```

Comment:
If we hide the `BuildEnvironment` model query behind the `ConnectionAwareLauncherProxy` and replace the fake `BuildConfiguration` with `GradleArguments` then we lose access to the `GradeEnvironment` here in the `BaseLaunchRequest` class (and won't be able to present the used Gradle version for instance). 

We could introduce the `GradleWorkspaceManager.getGradleBuild(GradleArguments)` but that seems way more work than it should be. Any ideas how to get around that?